### PR TITLE
kinetis: Merge cpu_conf settings into kinetis_common

### PR DESCRIPTION
**Pull request by gebart**
Another code deduplication PR.

This PR merges the settings which are common for all currently supported Kinetis CPUs into a single cpu_conf_kinetis.h file.

Based on #7395

### CODE REVIEW

File: cpu/kinetis_common/include/cpu_conf_kinetis.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * @name   ARM Cortex-M specific CPU configuration
+ * @{
+ */
+#define CPU_DEFAULT_IRQ_PRIO            (1U)
+/* Each interrupt priority setting is 8 bits wide, for both CM4 and CM0+, but
+ * they are laid out differently. CM0+ concatenates the settings into 32 bit
+ * words, CM4 and CM7 uses direct 8 bit access */
+#define CPU_IRQ_NUMOF                   (sizeof(NVIC->IP))
```
Comment (gebart) [inline cpu/kinetis_common/include/cpu_conf_kinetis.h]: I've tested this change on K22F, there's no error for setting priorities in the NVIC for interrupts which are not linked to anything internally, it only simplifies the configuration a lot.

**Comment by gebart** [comment]
Testing 
Testing status:

 - [x] frdm-k22f
 - [ ] frdm-k64f
 - [ ] mulle
 - [ ] pba-d-01-kw2x

@haukepetersen @aabadie @kYc0o do you have access to any of the above boards for testing?

**Comment by kYc0o** [comment]
I have a frdm-k64f right now and can access the others on monday.

**Comment by gebart** [comment]
@kYc0o do you mind testing #7379 as well on monday?

**Comment by kYc0o** [comment]
Yeah sure.

**Comment by gebart** [comment]
@kYc0o did you test this PR?

**Comment by kYc0o** [comment]
I'm on that. However I couldn't get a pba-d-01-kw2x...

**Comment by kYc0o** [comment]
Tests passed successfully. However I found a bug in the unittests for the mulle platform. I opened #7411 for tracking it. The unittests will anyways fail now in any platform but native.

**Comment by gebart** [comment]
@kYc0o do you know anyone who has a pba-d-01-kw2x board that we can ping for testing?

**Comment by kYc0o** [comment]
Maybe ping @haukepetersen @smlng

**Comment by smlng** [comment]
seems to work on `pba-d-01-kw2x` aka PhyNode, tested with `tests/periph_timer` and `tests/periph_gpio` - anything else? Otherwise, I'm fine with the changes and generally approve harmonising and simplifying board/cpu configurations 👍

**Comment by smlng** [comment]
however, do not merge, as this depends on #7395

**Comment by kYc0o** [comment]
#7395 just merged. Let's run Murdock again.

**Comment by gebart** [comment]
Murdock is green -> go

**Pull request merged**